archai/nlp/nvidia_transformer_xl/mem_transformer.py

```diff
@@ -1007,12 +1015,13 @@ def weights_init(m, weight_init_type:str, weight_init_range:float, weight_init_s
         if hasattr(m, 'bias') and m.bias is not None:
             init_bias(m.bias)
     elif classname.find('TransformerLM') != -1:
-        if hasattr(m, 'r_emb'):
-            init_weight(m.r_emb, **weight_init_params)
-        if hasattr(m, 'r_w_bias'):
-            init_weight(m.r_w_bias, **weight_init_params)
-        if hasattr(m, 'r_r_bias'):
-            init_weight(m.r_r_bias, **weight_init_params)
+        for i in range(m.n_layer):
+            if hasattr(m, f'r_emb_{i}'):
+                init_weight(getattr(m, f'r_emb_{i}'), **weight_init_params)
+            if hasattr(m, f'r_w_bias_{i}'):
```

Comment:
Review weight_init_params

Reply:
`weight_init_params` seems to be holding well. We are getting better baselines, at least during the first 20k steps. Will come back with 40k steps results.

Reply:
After 40k steps, we were able to improve the baseline by 0.75%.